src/main/java/com/google/cloud/genomics/utils/Contig.java

```diff
@@ -99,7 +99,7 @@ public Contig apply(String contigString) {
   // the ShardUtils methods should be used to ensure that shards are shuffled all together before
   // being returned to clients.
   List<Contig> getShards(long numberOfBasesPerShard) {
-    double shardCount = Math.ceil(end - start) / numberOfBasesPerShard;
+    double shardCount = (end - start) / (double) numberOfBasesPerShard;
```

Comment:
You can just keep them as long - the double has no effect, as the decimal portion and will not have impact in the for loop.  Any decimal portion in a long will be just Math.floor automatically.